syweb/webclient/app/settings/settings-controller.js

```diff
@@ -45,16 +45,76 @@ function($scope, matrixService, modelService, eventHandlerService, notificationS
     };
 
     function fetchRules() {
-        notificationService.getGlobalRulesets().then(function(rulesets) {
-            $scope.settings.rules = rulesets;
+        notificationService.getRulesets().then(function(rulesets) {
+            $scope.settings.rules = rulesets.global;
+
+            var rule_descriptions = {
+                '.m.rule.master': '',
+                '.m.rule.contains_user_name': "Notify me with sound about messages that contain my user name",
+                '.m.rule.contains_display_name': "Notify me with sound about messages that contain my display name",
+                '.m.rule.room_one_to_one': "Notify me with sound about messages sent just to me",
+                '.m.rule.suppress_notices': "Suppress notifications from bots",
+                '.m.rule.invite_for_me': "Notify me when I'm invited to a new room",
+                '.m.rule.member_event': "Notify me when people join or leave rooms",
+                '.m.rule.message': "Notify for all other messages/rooms",
+                '.m.rule.call': "Notify me when I receive a call",
+                '.m.rule.fallback': "Notify me for anything else"
+            };
+            var rule_categories = {
+                '.m.rule.master': 'master',
+                '.m.rule.suppress_notices': 'suppression',
+                '.m.rule.message': 'fallthrough',
+                '.m.rule.call': 'call',
+            };
+
+            var defaultRules = {master: [], additional: [], call: [], fallthrough: [], suppression: []};
+            for (var kind in rulesets.global) {
+                for (var i = 0; i < Object.keys(rulesets.global[kind]).length; ++i) {
+                    var r = rulesets.global[kind][i];
+                    var cat = rule_categories[r.rule_id];
+                    r.kind = kind;
+                    if (r.rule_id[0] == '.' && rule_descriptions[r.rule_id] !== undefined) {
+                        r.description = rule_descriptions[r.rule_id];
+                        if (cat) {
+                            defaultRules[cat].push(r);
+                        } else {
+                            defaultRules.additional.push(r);
+                        }
+                    }
+                }
+            }
+            $scope.settings.default_rules = defaultRules;
+            if (defaultRules.master.length > 0) {
+                $scope.settings.push_master_rule = defaultRules.master[0];
+            }
         });
     };
 
+    $scope.rule_add_input = {
+        content: '',
+        room: '',
+        sender: ''
+    };
     $scope.rule_add_action = {
         content: "notify",
         room: "notify",
         sender: "notify"
     };
+    $scope.rule_add_sound = {
+        content: null,
+        room: null,
+        sender: null
+    };
+    $scope.rule_highlight = {
+        content: false,
+        room: false,
+        sender: false
+    };
+    $scope.rule_add_inprogress = {
```

Comment:
This along with ``rule_add_sound``, ``rule_add_input`` and ``rule_highlight`` should all hang off the same object to avoid polluting the ``$scope`` variable so much, e.g.
```
$scope.rules = {
  add_sound: {
    content: null,
    ...
  },
  highlight: {
    ..
  }
```